### PR TITLE
Update Boolean and Text functions to have parity with TS

### PR DESCRIPTION
**Pull request by bryancroteau-MSFT**
Text(Blank()) will now return Blank() instead of ""
Boolean("True") => true, but Boolean(" true") => Error

### CODE REVIEW

**Review by lesaltzm — DISMISSED** [review]

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Boolean.txt [hunk 1]
```diff
@@ -5,28 +5,28 @@ true
...
 >> Boolean(" true")
-true
+false
 
 >> Boolean(" false")
 false
 
 >> Boolean("True")
-true
+false
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Boolean.txt:17]: This seems surprising...

**Review by MikeStall — DISMISSED** [review]
:clock1:

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs [hunk 2]
```diff
@@ -126,16 +126,7 @@ public static FormulaValue Boolean(IRContext irContext, StringValue[] args)
...
-            if (string.IsNullOrEmpty(str))
-            {
-                return new BlankValue(irContext);
-            }
-
-            if (!bool.TryParse(str, out var val))
-            {
-                return CommonErrors.InvalidBooleanFormatError(irContext);
-            }
+            var val = args[0].Value == "true";
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs:129]: >== "true" [](http://example.com/codeflow?start=36&length=9)

is this too strict?

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Boolean.txt (hunk 1 above)
Comment (bryancroteau-MSFT) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Boolean.txt:17]: This is how Canvas currently works.

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs (hunk 2 above)
Comment (bryancroteau-MSFT) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs]: Changed

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Boolean.txt (hunk 1 above)
Comment (bryancroteau-MSFT) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Boolean.txt]: Changed

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt [hunk 3]
```diff
@@ -131,10 +129,10 @@ Blank()
 
 //Excel needs two parameters for Text function
 >> Text(Blank())
-""
+Blank()
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt:132]: Text(<blank>) should return `""`

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Boolean.txt [hunk 4]
```diff
@@ -8,21 +8,27 @@ false
...
+
+>> Boolean("FALSE")
+false
+
 >> Boolean(Text(Blank()))
 Blank()
 
->> Boolean(Text(0/1))
+>> Boolean(Text(1/0))
 #Error
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Boolean.txt:32]: `#Error(Kind=Div0)` - if we know the error kind we should assert it as well

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt (hunk 3 above)
Comment (bryancroteau-MSFT) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text.txt]: Fixed

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/BasicCoercion.txt [hunk 5]
```diff
@@ -35,10 +35,10 @@ true
 
 // can coorce text to bool via logical operators
 >> "true" && "anything"
-false
+#Error
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/BasicCoercion.txt:38]: (Kind=InvalidArgument) - and can you also update the function InvalidBooleanFormatError in CommonErrors.cs to use this kind? Currently it uses 'BadLanguageCode', which is incorrect.

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/BasicCoercion.txt [hunk 6]
```diff
@@ -35,10 +35,10 @@ true
...
 
 >> "anything" || "true"
-true
+#Error
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/BasicCoercion.txt:41]: (Kind=InvalidArgument)

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Boolean.txt [hunk 7]
```diff
@@ -8,22 +8,28 @@ false
...
 
 >> Boolean(" true")
-true
+#Error
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Boolean.txt:11]: (Kind=InvalidArgument). Same for the other (runtime) errors when converting from string

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUnary.cs [hunk 8]
```diff
@@ -281,10 +281,26 @@ public static NumberValue BooleanToNumber(IRContext irContext, BooleanValue[] ar
...
-            var s = args[0].Value;
-            return new BooleanValue(irContext, s == "true");
+            var val = args[0].Value;
+
+            if (string.IsNullOrEmpty(val))
+            {
+                return new BlankValue(irContext);
+            }
+
+            var lower = val.ToLower();
```
[the remark this answers is not in the document]
Comment (bryancroteau-MSFT) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUnary.cs]: Done

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/BasicCoercion.txt (hunk 5 above)
Comment (bryancroteau-MSFT) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/BasicCoercion.txt]: done

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/BasicCoercion.txt (hunk 6 above)
Comment (bryancroteau-MSFT) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/BasicCoercion.txt]: done

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Boolean.txt (hunk 7 above)
Comment (bryancroteau-MSFT) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Boolean.txt]: done

**Review by CarlosFigueiraMSFT — APPROVED** [review]
:shipit:

**Pull request merged**